src/test/java/com/uniovi/main/selenium/WebTest.java

```diff
@@ -80,4 +81,14 @@ private void logInAs(AgentInfo agentInfo) {
 
 		driver.findElement(By.tagName("button")).click();
 	}
+	
+	@Test
+	public void checkLandingPage() {
+		driver.navigate().to(baseUrl+"/");
+		assertEquals(driver.getCurrentUrl(), baseUrl+"/agentForm");
```

Comment:
You are always redirecting to incidents in the controller:
```java
@RequestMapping(value="/", method=RequestMethod.GET)
public String landingPage() {
	return "redirect:/incidents";
}
```
Maybe that's why the test fails. You need to change the method to redirect to the agentForm if the agent is not logged in. I'm pretty sure SpringSecurity is not configured, so you can do as in other controller methods:
```java
Object info = session.getAttribute("agentInfo");
if (info == null) {
	return "redirect:/agentform";
}
```
If you want to setup spring security to make things cleaner, go ahead :wink:

Reply:
nevermind, if you redirect to incidents, it will be then redirected to agentform if needed. I don't like things to be so hidden/implicit, so maybe we should open an issue to setup Spring Security.